cpp/src/structure/detail/structure_utils.cuh

```diff
@@ -496,6 +498,62 @@ void sort_adjacency_list(raft::handle_t const& handle,
   }
 }
 
+template <typename T>
+struct indirect_array_reference {
+  T const* array_;
+
+  T operator() __host__ __device__(size_t index) { return array_[index]; }
+};
```

Comment:
Can't we use 
```
template <typename index_t, typename Iterator>
struct indirection_t {
  Iterator first{};

  __device__ typename thrust::iterator_traits<Iterator>::value_type operator()(index_t i) const
  {
    return *(first + i);
  }
};
```
instead?

https://github.com/rapidsai/cugraph/blob/branch-23.12/cpp/include/cugraph/utilities/device_functors.cuh#L70